src/kmer.py

```diff
@@ -1,31 +1,59 @@
-"""Computing kmers of a string."""
-
-
-def kmer(x: str, k: int) -> list[str]:
-    """
-    Computer all k-mers of x.
-
-    >>> kmer('agtagtcg', 3)
-    ['agt', 'gta', 'tag', 'agt', 'gtc', 'tcg']
-
-    FIXME: do you want more tests here?
-    """
-    ...
-
-
-def unique_kmers(x: str, k: int) -> list[str]:
-    """
-    Computer all unique k-mers of x.
-
-    FIXME: do you want more tests here?
-    """
-    ...
-
-
-def count_kmers(x: str, k: int) -> dict[str, int]:
-    """
-    Computer all k-mers of x and count how often they appear.
-
-    FIXME: do you want more tests here?
-    """
-    ...
+"""Computing kmers of a string."""
+
+
+def kmer(x: str, k: int) -> list[str]:
+    """
+    Computer all k-mers of x.
+
+    >>> kmer('agtagtcg', 3)
+    ['agt', 'gta', 'tag', 'agt', 'gtc', 'tcg']
+
+    >>> kmer('agtagtcg', 4)
+    ['agta', 'gtag', 'tagt', 'agtc', 'gtcg']
+
+    >>> kmer('agtagctagt', 0)
+    []
+    """
+    if k > 0:
+        kmers = [x[i:i + k] for i in range(len(x)-k+1)]
+    else:
+        return []
+    return kmers
+
+
+
+def unique_kmers(x: str, k: int) -> list[str]:
+    """
+    Computer all unique k-mers of x.
+
+    >>> unique_kmers('agtagtcg', 3)
+    ['agt', 'gta', 'tag', 'gtc', 'tcg']
+
+    >>> unique_kmers('gatcgatcgatc', 4)
+    ['gatc', 'atcg', 'tcga', 'cgat']
+    """
+    kmers = []

```

Comment:
Using a list here is somewhat inefficient. When you check if an element is in the list, you are doing a linear search. If you use a `set` instead, you can check if an element is there in constant time.